enrollment-server/src/main/java/com/wultra/app/enrollmentserver/mock/MockCustomActivationProvider.java

```diff
@@ -87,7 +89,21 @@ public String lookupUserIdForAttributes(Map<String, String> identityAttributes,
                 if (verifyResponse.isVerified()) {
                     return onboardingProcessService.getUserId(processId);
                 }
-                throw new PowerAuthActivationException();
+                if (verifyResponse.getOnboardingStatus() != OnboardingStatus.ACTIVATION_IN_PROGRESS) {
+                    // Onboarding process status is not ACTIVATION_IN_PROGRESS, onboarding process failed
+                    logger.info("Onboarding process failed during activation, process ID: {}, status: {}", processId, verifyResponse.getOnboardingStatus());
+                    throw new PowerAuthActivationException();
+                }
+                if (verifyResponse.getRemainingAttempts() > 0) {
+                    // There are remaining attempts for OTP verification
+                    logger.info("OTP verification attempt failed during activation, process ID: {}, remaining attempts: {}", processId, verifyResponse.getRemainingAttempts());
+                    final PowerAuthActivationOtpException ex = new PowerAuthActivationOtpException();
```

Comment:
dtto

Reply:
I will not do it right now, because during activation this is how activation exceptions are handled in restful-integration. We can probably change it one day, but we need to analyze it properly.